pkg/proof/README.md

```diff
@@ -0,0 +1,198 @@
+# Proof
+
+This package enables proof queries for Celestia transactions and blobs.
+
+
+## Prerequisites
+
+To understand the proof mechanisms,
+a good understanding of [binary merkle proofs](https://celestiaorg.github.io/celestia-app/specs/data_structures.html#merkle-trees) and [namespace merkle proofs](https://celestiaorg.github.io/celestia-app/specs/data_structures.html#namespace-merkle-tree) is required.
+
+## PFB transaction
+
+When creating a [PayForBlob](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31) transaction,
+the blob data is separated into a set of [shares](https://celestiaorg.github.io/celestia-app/specs/shares.html).
+This set of shares is used
+to generate a [*share commitment*](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html?highlight=share%20commitment#blob-share-commitment-rules)
+which commits to the data contained in the PFB,
+i.e., the blob.
+
+### Share commitment generation 
+
+Generating a *share commitment* starts with laying the shares into a [merkle mountain range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure.
+For example, if the blob contains six shares, the following structure will be generated:
+
+<img src="img/subtree_roots.png" alt="subtree roots generation from shares" width="500"/>
+
+The blob shares are colored in orange,
+and are used
+to generate a set of roots following the [merkle mountain range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure.
+These roots are called *subtree roots*.
+In the above case, we end up with two *subtree roots*: `SR1` and `SR2`.
+
+Then, these *subtree roots* are used to generate the *share commitment*:
+
+<img src="img/share_commitment.png" alt="share commitment generation from subtree roots" width="300"/>
+
+The *share commitment* is the binary merkle root over the set of *subtree roots* generated from the shares.
+
+## Share to share commitment inclusion
+
+Using the above merkle trees,
+share proofs can be generated to prove inclusion of a set of shares to the *share commitment*.
+
+For instance, if we want to prove the first two shares, we will need two merkle proofs:
+
+<img src="img/share_to_share_commitment_inclusion.png" alt="shares to share commitment inclusion proof" width="500"/>
+
+- The first merkle proof is a namespace merkle inclusion proof from the first two shares to the *subtree root* `SR1`.
+- The second merkle proof is a binary merkle inclusion proof from `SR1`,
+  which is the subtree root committing to the shares, to the *share commitment*.
+
+Note: the nodes colored in maroon are the inner nodes used to construct the inclusion proof.
```

Comment:
nit, I think it could be simpler if we just say red instead of maroon in this document. I could see folks having to look that color up since its less common

Reply:
[75c9054](https://github.com/celestiaorg/celestia-app/pull/3245/commits/75c90547fee1895afc95eca12540fce21d388b6b)